cohesive-vns3-free-multiclient-overlay-linux/azuredeploy.json

```diff
@@ -0,0 +1,485 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Admin User Name for VNS3 (required but not used)"
+            }    
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Admin Password for VNS3 (required but not used)"
+            }
+        },
+        "adminUsernameUbuntu": {
+            "type": "string",
+            "metadata": {
+                "description": "Admin User Name for Ubuntu VM"
+            }    
+        },
+        "adminPasswordUbuntu": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Admin Password for Ubuntu VM"
+            }
+        },
+        "numberOfInstances": {
+            "type": "int",
+            "minValue": 1,
+            "maxValue": 5,
+            "metadata": {
+              "description": "VMs to deploy, max 5 as free edition only supports 5 clientpacks"
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "Deployment location"
+            }
+        },
+        "instanceSize": {
+            "type": "string",
+            "metadata": {
+                "description": "Size of VM instance, defaults to Standard_B1ms to keep within core limits while giving reasonable performance"
+            },
+            "defaultValue": "Standard_B1ms"
+        }
+             
+    },
+    "variables": {
+        "suffix": "[uniqueString(resourceGroup().id, parameters('location'))]",
+        "resourceGroupName": "[toLower(resourceGroup().name)]",
+        "storageAccount": "[substring(concat(parameters('location'),'storage', variables('suffix')), 0, 24)]",
+        "virtualNetworkName": "[concat(variables('resourceGroupName'), '-vnet')]",
+        "vns3ControllerName": "VNS3Controller",
+        "networkInterfaceName": "[toLower(concat(variables('vns3ControllerName'), '-nic'))]",
+        "publicIPAddresseName": "[concat(variables('vns3ControllerName'), '-ip')]",
+        "networkSecurityGroupName": "[concat(variables('vns3ControllerName'), '-nsg')]",
+        "virtualSubnetName": "VNS3_VirtualSubnet",
+        "vnetSubnet":"10.10.10.0/28",
+        "vnsStaticIp": "10.10.10.10",
+        "CustomScriptForLinux": "CustomScript",
+        "artifactsLocation":"https://raw.githubusercontent.com/Cohesive-DevTeam/Quickstart/master/",
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts